packages/codec/lib/interface/decoders/contract.ts

```diff
@@ -79,53 +83,101 @@ export default class ContractDecoder {
     }
   }
 
+  /**
+   * @hidden
+   */
   public async init(): Promise<void> {
     this.contractNetwork = (await this.web3.eth.net.getId()).toString();
   }
 
+  /**
+   * Constructs a contract instance decoder for a given instance of this contract.
+   * @param address The address of the contract instance decode.  If left out, it will be autodetected.
+   */
   public async forInstance(address?: string): Promise<ContractInstanceDecoder> {
     let instanceDecoder = new ContractInstanceDecoder(this, address);
     await instanceDecoder.init();
     return instanceDecoder;
   }
 
+  /**
+   * Takes a Web3 Transaction object and returns a copy of that object but with
+   * an additional decoding field.  This field holds a CalldataDecoding; see the
+   * documentation on DecodedTransaction for more.
```

Comment:
(needs link)

Reply:
Similar question, although here the link would be across files (if that makes a difference).